tests/post_training/pipelines/lm_weight_compression.py

```diff
@@ -0,0 +1,208 @@
+        for line in stdout.splitlines():
+            for attr_name, prefix_regex in zip(self.VAR_NAMES, self.REGEX_PREFIX):
+                match = re.search(r"{}{}".format(prefix_regex, time_regex), line)
+                if match:
+                    setattr(self, attr_name, match.group(1))
+                continue
+
+    def get_stats(self) -> Dict[str, str]:
+        VARS = [getattr(self, name) for name in self.VAR_NAMES]
+        return dict(zip(self.STAT_NAMES, VARS))
+
+
+class LMWeightCompression(BaseTestPipeline):
+    """Pipeline for casual language models from Hugging Face repository"""
+
+    OV_MODEL_NAME = "openvino_model.xml"
+
+    def prepare_model(self) -> None:
+        if not (self.fp32_model_dir / self.OV_MODEL_NAME).exists():
+            # export by model_id
+            self.model_hf = OVModelForCausalLM.from_pretrained(
+                self.model_id, export=True, load_in_8bit=False, compile=False, stateful=False
+            )
+            self._dump_model_fp32()
+        else:
+            # no export, load from IR
+            self.model_hf = OVModelForCausalLM.from_pretrained(
+                self.fp32_model_dir, trust_remote_code=True, load_in_8bit=False, compile=False, stateful=False
+            )
+        self.model = self.model_hf.model
+
+    def prepare_preprocessor(self) -> None:
+        self.preprocessor = AutoTokenizer.from_pretrained(self.model_id)
+
+    def get_transform_calibration_fn(self):
+        def transform_fn(data):
+            tokenized_text = self.preprocessor(data["text"], return_tensors="np")
+            input_ids = tokenized_text["input_ids"]
+            attention_mask = tokenized_text["attention_mask"]
+
+            inputs = {}
+            inputs["input_ids"] = input_ids
+            inputs["attention_mask"] = tokenized_text["attention_mask"]
+            position_ids = np.cumsum(attention_mask, axis=1) - 1
+            position_ids[attention_mask == 0] = 1
+
+            # The magic forms KV cache as model inputs
+            batch_size = input_ids.shape[0]
+            for input_name in self.model_hf.key_value_input_names:
+                model_inputs = self.model.input(input_name)
+                shape = model_inputs.get_partial_shape()
+                shape[0] = batch_size
+                if shape[2].is_dynamic:
+                    shape[2] = 0
+                else:
+                    shape[1] = 0
+                inputs[input_name] = ov.Tensor(model_inputs.get_element_type(), shape.get_shape())
+
+            inputs["position_ids"] = position_ids
+            return inputs
+
+        return transform_fn
+
+    def prepare_calibration_dataset(self):
+        dataset = load_dataset("wikitext", "wikitext-2-v1", split="train")
+        dataset = dataset.filter(lambda example: len(example["text"]) > 80)
+        self.calibration_dataset = nncf.Dataset(dataset, self.get_transform_calibration_fn())
+
+    def cleanup_cache(self):
+        dir_with_cache = "model_cache"
```

Comment:
Add call cleanup_cache, it's need to clean torch cache, OVModelForCausalLM contain export torch model to OV by ov.convert_model.
https://github.com/openvinotoolkit/nncf/pull/2171#issuecomment-1749742211

Reply:
It's already there. https://github.com/openvinotoolkit/nncf/pull/2425/files#diff-5d01260f55336ab44f9ae6583b583333bdbd46271c48cf70862bb17c33d5c191R400-R408
I just gave a more general name.